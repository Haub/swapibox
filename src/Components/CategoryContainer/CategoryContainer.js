import React from 'react';
import './CategoryContainer.css';
import { Card } from '../Card/Card';
import PropTypes from 'prop-types';

export const CategoryContainer = ({stateArray, currentlyDisplayed}) => {
	
	const cardArray = stateArray.map(item => {
		return(
			<Card
				{...item}
				key={Date.now() * Math.random()}
				currentlyDisplayed = {currentlyDisplayed}
			/>)
	})

	return(
		<div className="Container">
			{cardArray}
		</div>

	)
}

CategoryContainer.propTypes = {
	stateArray: PropTypes.array
}