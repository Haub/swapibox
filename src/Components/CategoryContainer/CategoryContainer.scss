.Container {
	width: 100%;
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	align-items: space-between;
	justify-content: space-around;
	padding-top: 1rem;
}