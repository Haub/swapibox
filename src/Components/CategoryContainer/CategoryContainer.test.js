import React from 'react';
import { CategoryContainer } from './CategoryContainer';
import { shallow } from 'enzyme';

describe('CategoryContainer', () => {
  let wrapper;
  let mockCardsList;
  let mockToggleFavorite;

  beforeEach(() => {
    mockToggleFavorite = jest.fn();
    mockCardsList = [{name: 'Luke Skywalker', species: 'human', homeworld: 'Tatooine', population: '200000'}, 
                    {name: 'Leia Organa', species: 'human', homeworld: 'Alderaan', population: '2000000000'}]
    wrapper = shallow(<CategoryContainer />);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it('has the correct number of groceries', () => {
    wrapper = shallow(
      <Card
        item={[
          {name: 'Luke Skywalker', species: 'human', homeworld: 'Tatooine', population: '200000'},
          {name: 'Leia Organa', species: 'human', homeworld: 'Alderaan', population: '2000000000'}
        ]}
      />
    );

    expect(wrapper.find('Card').length).toEqual(2);
  });

describe('toggleFavorite', () => {

  it('triggers toggleFavorite when clicked', () => {
    const wrapper = mount(
      <Card
        item={[
          {name: 'Luke Skywalker', species: 'human', homeworld: 'Tatooine', population: '200000'},
          {name: 'Leia Organa', species: 'human', homeworld: 'Alderaan', population: '2000000000'}
        ]}
      />
    );
    wrapper.find('.Card').simulate('click');
    expect(mockToggleFavorite).toBeCalled();
  });
});