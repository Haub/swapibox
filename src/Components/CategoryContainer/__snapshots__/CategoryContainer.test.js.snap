// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryContainer matches the snapshot 1`] = `undefined`;
